File: nvxapp.client/src/nvx/ClientServer-Service/GestionePresenze/VacationPlanService/Models/vacation-plan-model.ts
import { ModelResult } from "../../../ModelsBase/model-result";

export class VacationPlanModel {
  public year: number             = 0;
  public month: number            = 0;
  public selectedUserId: string[] = [];
}

export class VacationPlan_DaySlot {
  public idAspNetUsers: string = '';
  public isPresent: boolean    = false;
}

export class VacationPlan_GetInModel {
  public vacationPlan: VacationPlanModel = new VacationPlanModel();
}

export class VacationPlan_GetOutModel extends ModelResult {
  public daySlots: VacationPlan_DaySlot[] = [];
}

File: nvxapp.server.service/ClientServer-Service/GestionePresenze/VacationPlanService/Models/VacationPlanModel.cs
using nvxapp.server.service.ClientServer_Service.GestionePresenze.Dip_GG_GiustificativiService.Models;
using nvxapp.server.service.ClientServer_Service.GestionePresenze.Dip_GG_RichiestaService.Models;
using nvxapp.server.service.ClientServer_Service.GestionePresenze.Dip_GG_TimbraturaService.Models;
using nvxapp.server.service.ClientServer_Service.GestionePresenze.PresentStaffService.Models;
using nvxapp.server.service.ClientServer_Service.ModelsBase;

namespace nvxapp.server.service.ClientServer_Service.GestionePresenze.VacationPlanService.Models
{
    public class VacationPlanModel
    {
        public int Year { get; set; }
        public int Month { get; set; }
        public List<string> SelectedUserId { get; set; } = new List<string>();
    }

    public class VacationPlan_DaySlot
    {
        public string IdAspNetUsers { get; set; } = string.Empty;
        public List<Dip_GG_RichiestaModel> Dip_GG_Richieste { get; set; } = new List<Dip_GG_RichiestaModel>();
        public List<Dip_GG_GiustificativiModel> Dip_GG_Giustificativi { get; set; } = new List<Dip_GG_GiustificativiModel>();
    }

    public class VacationPlan_GetInModel
    {
        public VacationPlanModel VacationPlan { get; set; } = new VacationPlanModel();
    }

    public class VacationPlan_GetOutModel : ModelResult
    {
        public List<VacationPlan_DaySlot> DaySlots { get; set; } = new List<VacationPlan_DaySlot>();
    }
}
